### PR TITLE
Add preserve_colortype parameter in convertToGrayscale

### PR DESCRIPTION
**Pull request by manorit2001**

### CODE REVIEW

File: include/CIL/Transformations.hpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ namespace CIL {
     void
     rotate(ImageInfo& img, int degrees,
            RotationKind rotation_kind = RotationKind::rotation_by_area_mapping);
-    void convertToGrayscale(ImageInfo& img);
+    void convertToGrayscale(ImageInfo& img, bool preserve_components = true);
```
Comment (parth-07) [inline include/CIL/Transformations.hpp:44]: Can you please change `preserve_componenets` to `preserve_colortype`? `preserve_colortype` seems more intuitive.

**Comment by parth-07** [comment]
Please also add `grayscale` test information in `tests/Makefile.am` `check_PROGRAMS`.

**Comment by manorit2001** [comment]
fixed both of them

**Review by parth-07 — APPROVED** [review]
LGTM.

**Pull request merged**